common-user: Only compile the common user code if have_user is set

Index: common-user/meson.build
--- common-user/meson.build.orig
+++ common-user/meson.build
@@ -1,3 +1,7 @@
+if not have_user
+   subdir_done()
+endif
+
 common_user_inc += include_directories('host/' / host_arch)
 
 user_ss.add(files(
